.github/workflows/snapcraft-candidate.yml

```diff
@@ -0,0 +1,29 @@
+name: snapcraft-candidate
+# Builds and publishes the package to the candidate channel on merge to main.
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build-and-publish:
+    name: 'Build and publish'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
```

Comment:
Even though we are not publishing a Docker image in this one, we still need to do the Docker login dance. `make _build_snap` pull the build container and Docker is now rate limiting pulls. Since Travis shares IPs across builds we were hitting this even though we never come close to that ourselves.

See: https://github.com/digitalocean/doctl/pull/941

Looks like this is a concern for GitHub Actions as well:

https://docs.docker.com/docker-hub/download-rate-limit/#github-actions